packages/lib-utils/src/k8s/hooks/useK8sWatchResource.ts

```diff
@@ -76,6 +83,10 @@ export const useK8sWatchResource = <R extends K8sResourceCommon | K8sResourceCom
     }
     if (!resourceK8s) {
       const data = resource.isList ? [] : {};
+      if (!k8sModel && !inFlight && !batchesInFlight && retries < 3) {
+        fetchModel(resource, dispatch);
+        setRetries(retries + 1);
+      }
       return modelsLoaded && !k8sModel && !batchesInFlight
         ? [data, true, new NoModelError()]
```

Comment:
We may need to prevent this return value from returning while we're still trying to fetch the model on demand above. While I was testing locally this didn't _seem_ to be an issue, but I'll continue looking.